operator/standalone/update.go

```diff
@@ -0,0 +1,386 @@
+			"existingSecret":     getCredentialSecretName(),
+			"database":           u.instance.Name,
+			"username":           u.instance.Name,
+		},
+		"primary": helmvalues.V{
+			"resources": helmvalues.V{
+				"limits": helmvalues.V{
+					"memory": u.instance.Spec.Parameters.Resources.MemoryLimit.String(),
+				},
+			},
+			"persistence": helmvalues.V{
+				"size": u.instance.Spec.Parameters.Resources.StorageCapacity.String(),
+			},
+		},
+		"fullnameOverride": getDeploymentName(),
+		"networkPolicy": helmvalues.V{
+			"enabled": true,
+			"ingressRules": helmvalues.V{
+				"primaryAccessOnlyFrom": helmvalues.V{
+					"enabled": true,
+					"namespaceSelector": helmvalues.V{
+						"kubernetes.io/metadata.name": u.instance.Namespace,
+					},
+				},
+			},
+		},
+	}
+	helmvalues.Merge(resources, &u.helmValues)
+	return nil
+}
+
+// ensureHelmRelease updates the Helm release object
+func (u *UpdateStandalonePipeline) ensureHelmRelease(ctx context.Context) error {
+	chart := &u.config.Spec.HelmReleaseTemplate.Chart
+	helmValues, err := helmvalues.Marshal(u.helmValues)
+	if err != nil {
+		return err
+	}
+	u.helmRelease = &helmv1beta1.Release{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   u.instance.Status.HelmChart.DeploymentNamespace,
+			Labels: getCommonLabels(u.instance.Name),
+		},
+	}
+
+	updateRelease := func() error {
+		u.helmRelease.Spec = helmv1beta1.ReleaseSpec{
+			ForProvider: helmv1beta1.ReleaseParameters{
+				Chart:               helmv1beta1.ChartSpec{Repository: chart.Repository, Name: chart.Name, Version: chart.Version},
+				Namespace:           u.instance.Status.HelmChart.DeploymentNamespace,
+				SkipCreateNamespace: true,
+				SkipCRDs:            true,
+				Wait:                true,
+				ValuesSpec:          helmv1beta1.ValuesSpec{Values: helmValues},
+			},
+			ResourceSpec: crossplanev1.ResourceSpec{
+				ProviderConfigReference: &crossplanev1.Reference{Name: u.config.Spec.HelmProviderConfigReference},
+			},
+		}
+		return nil
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, u.client, u.helmRelease, updateRelease)
+	return err
+}
+
+func (u *UpdateStandalonePipeline) fetchOperatorConfig(ctx context.Context) error {
+	list := &v1alpha1.PostgresqlStandaloneOperatorConfigList{}
+	labels := client.MatchingLabels{
+		v1alpha1.PostgresqlMajorVersionLabelKey: u.instance.Spec.Parameters.MajorVersion.String(),
+	}
+	ns := client.InNamespace(u.operatorNamespace)
+	err := u.client.List(ctx, list, labels, ns)
+	if err != nil {
+		return err
+	}
+	if len(list.Items) == 0 {
+		return fmt.Errorf("no %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labels, ns)
+	}
+	if len(list.Items) > 1 {
+		return fmt.Errorf("multiple versions of %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labels, ns)
+	}
+	u.config = &list.Items[0]
+	return nil
+}
+
+func (u *UpdateStandalonePipeline) ensureConnectionSecret(ctx context.Context) error {
```

Comment:
Same here, IMHO there should be one idempotent `ensureConnectionSecret` that can be used for update and creation.

Reply:
Same here, we will do a refactor with a new PR.